django/db/models/constraints.py

```diff
@@ -1,54 +1,100 @@
 from django.db.models.sql.query import Query
 
-__all__ = ['CheckConstraint']
+__all__ = ['CheckConstraint', 'UniqueConstraint']
 
 
-class CheckConstraint:
-    def __init__(self, constraint, name):
-        self.constraint = constraint
+class BaseConstraint:
+    def __init__(self, name):
         self.name = name
 
     def constraint_sql(self, model, schema_editor):
-        query = Query(model)
-        where = query.build_where(self.constraint)
-        connection = schema_editor.connection
-        compiler = connection.ops.compiler('SQLCompiler')(query, connection, 'default')
-        sql, params = where.as_sql(compiler, connection)
-        params = tuple(schema_editor.quote_value(p) for p in params)
-        return schema_editor.sql_check % {
-            'name': schema_editor.quote_name(self.name),
-            'check': sql % params,
-        }
+        raise NotImplementedError
 
     def create_sql(self, model, schema_editor):
         sql = self.constraint_sql(model, schema_editor)
-        return schema_editor.sql_create_check % {
+        return schema_editor.sql_create_constraint % {
             'table': schema_editor.quote_name(model._meta.db_table),
-            'check': sql,
+            'name': schema_editor.quote_name(self.name),
+            'constraint': sql,
         }
 
     def remove_sql(self, model, schema_editor):
         quote_name = schema_editor.quote_name
-        return schema_editor.sql_delete_check % {
+        return schema_editor.sql_delete_constraint % {
             'table': quote_name(model._meta.db_table),
             'name': quote_name(self.name),
         }
 
+    def deconstruct(self):
+        path = '%s.%s' % (self.__class__.__module__, self.__class__.__name__)
+        path = path.replace('django.db.models.constraints', 'django.db.models')
+        return (path, (), {'name': self.name})
+
+    def clone(self):
+        _, args, kwargs = self.deconstruct()
+        return self.__class__(*args, **kwargs)
+
+
+class CheckConstraint(BaseConstraint):
+    def __init__(self, *, check, name):
```

Comment:
The motivation behind this signature change it detailed in the commit message but I wanted to gather feedback about it here.

1. Using keyword only arguments aligns this API with the `Index` one and makes sure arguments are explicitly named. This is something that was mentioned during the review of #7615 related to the ordering of `constraint` and `name`.
2. Changed the `constraint` name to `check` because it wasn't the actual constraint that was being provided. Maybe unnecessary.

Reply:
1: I'm definitely happy with this. I'm not sure why I didn't go for it myself.
2: Fine by me, I don't feel strongly.